bin/utils/test_versions_ops

```diff
@@ -0,0 +1,199 @@
+			fi
+			if [[ $update_test_versions -eq 0 ]]; then
+				printf '%32s%10s%10s%6s\n' $test_repo $using_version $latest $version_ok
+				if [[ $version_fail != "" ]]; then
+					echo Failure: $version_fail
+				fi
+			else
+				if [[ $version_ok == "no" ]]; then
+					#
+					# We have to walk the file instead of using a series of sed pipes.
+					# The reasons for this are:
+					#  1) We can not simply update the version number through the whole
+					#     file, we might change something we should not.  For example,
+					#     if an option passed to the test happens to match the current
+					#     version number in the file.
+					#       Example:
+					#         exec_dir: "test-wrapper-1.0/test"
+					#         repo_file: "v1.0.zip"
+					#         test_specific: "--option burden_version_1.0
+					#  2) If we do a full sed replacement, if there are extra spaces
+					#     in the line, we will not change it.
+					#  3) We could pull the required string via grep, and build the new
+					#     one.  However that will require reading the file 3 times (2
+					#     greps, and then one update), versus the one read if we simply
+					#     walk the file, and look for the line to replace.
+					#
+					echo Updating version for $test_repo from $using_version to $latest
+					while IFS= read -r replace_line
+					do
+						if [[ $replace_line == "repo_file:"* ]]; then
+							echo $replace_line | sed "s/$using_version/$latest/g" >> ${test_repo}.new
+							continue
+						fi
+						if [[ $replace_line == "exec_dir:"* ]]; then
+							replace_with=`echo $latest | sed "s/^v/-/g"`
+							replacing=`echo $using_version | sed "s/^v/-/g"`
+							echo $replace_line | sed "s/$replacing/$replace_with/g" >> ${test_repo}.new
+							continue
+						fi
+						echo $replace_line >> ${test_repo}.new
+					done < "${test_repo}"
+					mv $test_repo ${test_repo}.orig
+					mv ${test_repo}.new ${test_repo}
+				fi
+			fi
+			rm -rf $temp_dir 
+		else
+			echo $test_repo, Only github is supported.
+		fi
+	done
+	popd > /dev/null
+}
+
+ARGUMENT_LIST=(
+	"config_dir"
+	"test_version_check"
+	"update_test_versions"
+)
+
+NO_ARGUMENTS=(
+	"usage"
+)
+
+# read arguments
+opts=$(getopt \
+	--longoptions "$(printf "%s:," "${ARGUMENT_LIST[@]}")" \
+	--longoptions "$(printf "%s," "${NO_ARGUMENTS[@]}")" \
+	--name "$(basename "$0")" \
+	--options "h" \
+	-- "$@"
+)
+
+
+eval set --$opts
+if [[ $? -ne 0 ]]; then
+	usage $0
+fi
+
+
+while [[ $# -gt 0 ]]; do
+	case "$1" in
+		--config_dir)
+			config_dir=$2
+			shift 2
+		;;
+		--test_version_check)
+			test_version_check=$2
+			shift 2
+		;;
+		--update_test_versions)
+			update_test_versions=$2
+			shift 2
+		;;
+		--usage)
+			usage $0
+		;;
+		--h)
+			usage $0
+		;;
+		--)
+			break;
+		;;
+		*)
+		echo "option not found ${1}"
+			usage $0
+		;;
+	esac
+done
+
+test_version_ops
+
+if [ $update_test_versions -eq 1 ]; then
+	source ${UTILS_DIR}/cleanup_and_exit_out --msg_string "Update of test versions completed" --rtc 0
```

Comment:
Are we intending to kill off the parent shell?  This script is `source`'d in, which will inherit the parent shell, sourcing `cleanup_and_exit_out` with rtc 0 will kill the entire shell.

Reply:
It kills the caller, not the shell this is executed from (just verified).

bin/utils/test_versions_ops ....
does not kill the shell we ran from.  We need this so burden will exit out when it is suppose to.